File: AULA 11 - 13-10-25/POO_Aula11d_ListaExercicios_Poo_Parte3_CadastroProdutos/Q2_Produtos_Persistencia/Produto.java
/**
 * Classe que representa um Produto com todos os atributos solicitados
 * 
 * @author Professor Marçal
 * @version 2025
 */
public class Produto {
    // Atributos privados da classe Produto
    private int codigo;              // Código do produto
    private String nome;             // Nome do produto
    private String descricao;        // Descrição do produto
    private int quantidadeEstoque;   // Quantidade em estoque
    private double precoCusto;       // Preço de custo
    private double margemLucro;      // Margem de lucro (em percentual)

    /**
     * Construtor padrão da classe Produto
     */
    public Produto() {
        this.codigo = 0;
        this.nome = "";
        this.descricao = "";
        this.quantidadeEstoque = 0;
        this.precoCusto = 0.0;
        this.margemLucro = 0.0;
    }

    /**
     * Construtor parametrizado da classe Produto
     * 
     * @param codigo Código do produto
     * @param nome Nome do produto
     * @param descricao Descrição do produto
     * @param quantidadeEstoque Quantidade em estoque
     * @param precoCusto Preço de custo
     * @param margemLucro Margem de lucro
     */
    public Produto(int codigo, String nome, String descricao, int quantidadeEstoque, 
                   double precoCusto, double margemLucro) {
        this.codigo = codigo;
        this.nome = nome;
        this.descricao = descricao;
        this.quantidadeEstoque = quantidadeEstoque;
        this.precoCusto = precoCusto;
        this.margemLucro = margemLucro;
    }

    // Métodos getters e setters para acessar os atributos privados

    public int getCodigo() {
        return codigo;
    }

    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public int getQuantidadeEstoque() {
        return quantidadeEstoque;
    }

    public void setQuantidadeEstoque(int quantidadeEstoque) {
        this.quantidadeEstoque = quantidadeEstoque;
    }

    public double getPrecoCusto() {
        return precoCusto;
    }

    public void setPrecoCusto(double precoCusto) {
        this.precoCusto = precoCusto;
    }

    public double getMargemLucro() {
        return margemLucro;
    }

    public void setMargemLucro(double margemLucro) {
        this.margemLucro = margemLucro;
    }

    /**
     * Método para calcular o preço de venda
     * Calculado automaticamente: Preço Custo * Margem Lucro
     * @return Preço de venda calculado
     */
    public double getPrecoVenda() {
        return precoCusto * margemLucro;
    }

    /**
     * Método para exibir todas as informações do produto
     */
    public void exibirDados() {
        System.out.println("=== DADOS DO PRODUTO ===");
        System.out.println("Código: " + codigo);
        System.out.println("Nome: " + nome);
        System.out.println("Descrição: " + descricao);
        System.out.println("Quantidade em Estoque: " + quantidadeEstoque);
        System.out.printf("Preço de Custo: R$ %.2f%n", precoCusto);
        System.out.printf("Margem de Lucro: %.2f%%%n", margemLucro * 100);
        System.out.printf("Preço de Venda: R$ %.2f%n", getPrecoVenda());
        System.out.println("========================");
    }
}

File: AULA 10 - 06-10-25/POO_Aula10b_ListaExercicios_Poo_Parte1/Q3_Produtos/Produto.cs
using System;

namespace POO_Aula10b_ListaExercicios_Poo_Parte1
{
    /// <summary>
    /// Classe que representa um Produto com todos os atributos solicitados
    /// </summary>
    public class Produto
    {
        // Atributos privados da classe Produto
        private int codigo;              // Código do produto
        private string nome;             // Nome do produto
        private string descricao;        // Descrição do produto
        private int quantidadeEstoque;   // Quantidade em estoque
        private double precoCusto;       // Preço de custo
        private double margemLucro;      // Margem de lucro (em percentual)

        // Propriedades públicas para acessar os atributos privados
        /// <summary>
        /// Propriedade para acessar e modificar o Código do produto
        /// </summary>
        public int Codigo
        {
            get { return codigo; }
            set { codigo = value; }
        }

        /// <summary>
        /// Propriedade para acessar e modificar o Nome do produto
        /// </summary>
        public string Nome
        {
            get { return nome; }
            set { nome = value; }
        }

        /// <summary>
        /// Propriedade para acessar e modificar a Descrição do produto
        /// </summary>
        public string Descricao
        {
            get { return descricao; }
            set { descricao = value; }
        }

        /// <summary>
        /// Propriedade para acessar e modificar a Quantidade em Estoque
        /// </summary>
        public int QuantidadeEstoque
        {
            get { return quantidadeEstoque; }
            set { quantidadeEstoque = value; }
        }

        /// <summary>
        /// Propriedade para acessar e modificar o Preço de Custo
        /// </summary>
        public double PrecoCusto
        {
            get { return precoCusto; }
            set { precoCusto = value; }
        }

        /// <summary>
        /// Propriedade para acessar e modificar a Margem de Lucro
        /// </summary>
        public double MargemLucro
        {
            get { return margemLucro; }
            set { margemLucro = value; }
        }

        /// <summary>
        /// Propriedade somente leitura para o Preço de Venda
        /// Calculado automaticamente: Preço Custo * Margem Lucro
        /// </summary>
        public double PrecoVenda
        {
            get { return precoCusto * margemLucro; }
        }

        /// <summary>
        /// Construtor padrão da classe Produto
        /// </summary>
        public Produto()
        {
            codigo = 0;
            nome = "";
            descricao = "";
            quantidadeEstoque = 0;
            precoCusto = 0.0;
            margemLucro = 0.0;
        }

        /// <summary>
        /// Construtor parametrizado da classe Produto
        /// </summary>
        /// <param name="codigo">Código do produto</param>
        /// <param name="nome">Nome do produto</param>
        /// <param name="descricao">Descrição do produto</param>
        /// <param name="quantidadeEstoque">Quantidade em estoque</param>
        /// <param name="precoCusto">Preço de custo</param>
        /// <param name="margemLucro">Margem de lucro</param>
        public Produto(int codigo, string nome, string descricao, int quantidadeEstoque, double precoCusto, double margemLucro)
        {
            this.codigo = codigo;
            this.nome = nome;
            this.descricao = descricao;
            this.quantidadeEstoque = quantidadeEstoque;
            this.precoCusto = precoCusto;
            this.margemLucro = margemLucro;
        }

        /// <summary>
        /// Método para exibir todas as informações do produto
        /// </summary>
        public void ExibirDados()
        {
            Console.WriteLine("=== DADOS DO PRODUTO ===");
            Console.WriteLine($"Código: {Codigo}");
            Console.WriteLine($"Nome: {Nome}");
            Console.WriteLine($"Descrição: {Descricao}");
            Console.WriteLine($"Quantidade em Estoque: {QuantidadeEstoque}");
            Console.WriteLine($"Preço de Custo: R$ {PrecoCusto:F2}");
            Console.WriteLine($"Margem de Lucro: {MargemLucro:P2}");
            Console.WriteLine($"Preço de Venda: R$ {PrecoVenda:F2}");
            Console.WriteLine("========================");
        }
    }
}

